app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,205 @@
+        message: messages.resourceRequired
+      };
+    };
+
+    var showInvalidResource = function() {
+      $scope.alerts.invalidResource = {
+        type: "error",
+        message: messages.invalidResource
+      };
+    };
+
+    var namespaceWhitelist = window.OPENSHIFT_CONSTANTS.CREATE_FROM_URL_WHITELIST;
+    var whiteListedCreateDetailsKeys = ['namespace', 'app', 'imageStream', 'imageTag', 'sourceURI', 'sourceRef', 'contextDir', 'template', 'templateParamsMap'];
+    var createDetails = _.pick($routeParams, function(value, key) {
+      // routeParams without a value (e.g., ?app&) return true, which results in "true" displaying in the UI
+      return _.contains(whiteListedCreateDetailsKeys, key) && (value && value !== true);
+    });
+    // if no namespace is specified, set it to 'openshift'
+    createDetails.namespace = createDetails.namespace || 'openshift';
+
+    var getResources = function() {
+      if (createDetails.imageStream) {
+        DataService
+          .get("imagestreams", createDetails.imageStream, {namespace: createDetails.namespace}, {
+            errorNotification: false
+          })
+          .then(function(imageStream) {
+            DataService
+              .get("imagestreamtags", imageStream.metadata.name + ":" + createDetails.imageTag, {namespace: createDetails.namespace}, {
+                  errorNotification: false
+                })
+                .then(function(imageStreamTag){
+                  $scope.validationPassed = true;
+                  $scope.resource = imageStreamTag;
+                  createDetails.displayName = $filter('displayName')(imageStreamTag);
+                }, function(e){
+                  alertInvalidImageTag(createDetails.imageTag, e);
+                });
+          }, function(e) {
+            if(e.status === 403) {
+              alertInaccessibleNamespace(createDetails.namespace, e);
+            } else {
+              alertInvalidImageStream(createDetails.imageStream, e);
+            }
+          });
+      }
+      if (createDetails.template) {
+        DataService
+          .get("templates", createDetails.template, {namespace: createDetails.namespace}, {
+            errorNotification: false
+          })
+          .then(function(template) {
+            $scope.validationPassed = true;
+            $scope.resource = template;
+          }, function(e) {
+            if(e.status === 403) {
+              alertInaccessibleNamespace(createDetails.namespace, e);
+            } else {
+              alertInvalidTemplate(createDetails.template, e);
+            }
+          });
+      }
+    };
+
+    if (!(_.includes(namespaceWhitelist, createDetails.namespace))) {
+      alertInvalidNamespace(createDetails.namespace);
+    } else {
+      if (createDetails.imageStream && createDetails.template) {
+        showInvalidResource();
+      } else if (!(createDetails.imageStream) && !(createDetails.template)) {
+        alertResourceRequired();
```

Comment:
> Is the rest of the page blank? How does that look?

Yep.  It looks like a blank page.  I wasn't sure what else to include since it's hard to know what the user is trying to do.  Thoughts?